### PR TITLE
cilium: Add workflows for GKE in tunnel mode, with and without encryption

### PR DESCRIPTION
**Pull request by jrfastab**
We want to test vxlan with encryption add workflow tests to GKE to capture this use case.

### CODE REVIEW

**Comment by jrfastab** [comment]
This looks like an unrelated flake, it happens on the first connectivity test before encryption or tunnelling. https://github.com/cilium/cilium/pull/15678/checks?check_run_id=2339277155
```
✅ Drop not found for pod cilium-test/client-6c46bff775-p27ht
❌ RST tcpflags(ack,rst) found for pod cilium-test/client-6c46bff775-p27ht
📄 Flow logs of pod cilium-test/client-6c46bff775-p27ht:
Apr 14 03:42:33.372: 10.16.1.3:41097 -> 10.16.1.212:53 from-endpoint FORWARDED 
```

File: .github/workflows/gke.yaml
Comment (nbusseneau) [inline .github/workflows/gke.yaml:208]: I think we're missing something here... 😇 
```suggestion
            --encryption \
            --cluster-name=${{ env.clusterName }} \
```

**Review by nbusseneau — COMMENTED** [review]
[[Link to Cilium Slack thread to discuss this]](https://cilium.slack.com/archives/C7PE7V806/p1618406776180900)

To be honest I am a bit concerned... At first we had a simple, straightforward workflow as in "just run `cilium install` and then `cilium connectivity test`, boom", with some fluff around for artifacts etc.

Then we added another set for encryption. Now we're adding two more for tunnel datapath mode.

My concern is: when does it end? 🤔 
- Is it really what we want to do and move forward with regarding platform testing via CI 3.0?
- Would it make sense to have separate workflows for separate concerns, rather than bundling four (!) `cilium install` / `cilium connectivity test` in a single workflow?
- Would it make sense to ditch the current `cilium install` / `cilium connectivity test` approach and return to the drawing board so that a single `cilium install` / `cilium connectivity test` (perhaps with special CLI parameters) does them all?

If we keep the current approach and do want to bundle these in a single workflow, then I will make a note of doing a refactoring pass in order to reduce the amount of duplication in here.

Disclaimer: I'm not a datapath guy, so I have literally zero idea what makes sense or not.

File: .github/workflows/gke.yaml
Comment (jrfastab) [reply .github/workflows/gke.yaml:208]: oops good catch I've add this now.

**Comment by jrfastab** [comment]
The conformanceGKE test missed the initial syn/ack? I bumped the sleep after the port forwarding enable from 5s to 10s.

**Comment by jrfastab** [comment]
bumped into timeout while copying logs but all GKE tests ran and passed so lets bump to 25minutes.

File: .github/workflows/gke.yaml
Comment (nbusseneau) [inline .github/workflows/gke.yaml]: I think we can actually just drop all the `--restart-unmanaged-pods=true` flags since that's the default anyway.

**Review by nbusseneau — APPROVED** [review]

File: .github/workflows/gke.yaml
Comment (jrfastab) [reply .github/workflows/gke.yaml]: ah yeah lets do it.

**Comment by jrfastab** [comment]
Lets run one more time to be sure I didn't typo anything while deleting unmanaged-pods.

**Comment by jrfastab** [comment]
This is stable now and passing ConformanceGKE consistently I'm going to drop the test commit now and mark it ready.

**Comment by jrfastab** [comment]
Seeing this is a workflow and does not touch other cilium code I see no reason to run tests. Marking ready to merge, there is the direction question from @nbusseneau above though. In my opinion we want to merge this, it bumps the max time from 15minutes to 20minutes, but without this we have no tests for encryption with vxlan that cover these cases. In the past we've introduced regressions in this space that have resulted in a fair amount of work to unwind so from my perspective the extra testing is worth the complexity/cost.

**Comment by qmonnet** [comment]
Janitor here: The PR has been in draft mode so far, so few people took the time to review. I'd rather have it marked as “ready for review” for the time being and gather more feedback, as the concerns from Nicolas sound legitimate. Plus we're missing a review from cilium/maintainers.

I'll clean up the labels and draft status for now.

**Comment by qmonnet** [comment]
>  it bumps the max time from 15minutes to 20minutes

30 minutes in the YAML file

**Comment by nbusseneau** [comment]
@qmonnet Thomas gave his opinion [here](https://cilium.slack.com/archives/C7PE7V806/p1618411186187100?thread_ts=1618406776.180900&cid=C7PE7V806), so I think we are aligned. Let's just merge as-is for now, and I will make a refactoring pass to switch to a matrix-based workflow later.

**Review by aanm — APPROVED** [review]

**Pull request merged**
